TrackingProduction/Fun4All_FieldOnAllTrackersCalos.C

```diff
@@ -0,0 +1,411 @@
+R__LOAD_LIBRARY(libmvtx.so)
+R__LOAD_LIBRARY(libintt.so)
+R__LOAD_LIBRARY(libtpc.so)
+R__LOAD_LIBRARY(libmicromegas.so)
+R__LOAD_LIBRARY(libTrackingDiagnostics.so)
+R__LOAD_LIBRARY(libtrack_reco.so)
+R__LOAD_LIBRARY(libtrack_to_calo.so)
+R__LOAD_LIBRARY(libcalo_reco.so)
+
+using namespace std;
+
+std::string GetFirstLine(std::string listname);
+
+void Fun4All_FieldOnAllTrackersCalos(
+    const int nEvents = 10, 
+    vector<string> myInputLists = {
+        "run46730_0000_trkr.txt",
+        "run46730_calo.list"}, 
+    bool doTpcOnlyTracking = true,
+    bool doDriftVelocityCorr = true,
+    bool doEMcalRadiusCorr = true,
+    const bool convertSeeds = false)
+{
+  int verbosity = 0;
+
+  gSystem->Load("libg4dst.so");
+  //gSystem->Load("libFROG.so");
+  //FROG *fr = new FROG();
+
+  std::string firstfile = GetFirstLine(myInputLists[0]);
+  if (*firstfile.c_str() == '\0') return;
+  std::pair<int, int> runseg = Fun4AllUtils::GetRunSegment(firstfile);
+  int runnumber = runseg.first;
+  int segment = runseg.second;
+
+  auto rc = recoConsts::instance();
+  rc->set_IntFlag("RUNNUMBER", runnumber);
+
+  Enable::CDB = true;
+  rc->set_StringFlag("CDB_GLOBALTAG", "ProdA_2024");
+  rc->set_uint64Flag("TIMESTAMP", 6);
+
+  string outDir = "./";
+
+  Enable::DSTOUT = false;
+
+  //Create the server
+  Fun4AllServer *se = Fun4AllServer::instance();
+  se->Verbosity(verbosity);
+
+  std::cout << "Including " << myInputLists.size() << " files." << std::endl;
+
+  //Add all required input files
+  for (unsigned int i = 0; i < myInputLists.size(); ++i)
+  {
+    Fun4AllInputManager *infile = new Fun4AllDstInputManager("DSTin_" + to_string(i));
+    std::cout << "Including file " << myInputLists[i] << std::endl;
+    //infile->AddFile(myInputLists[i]);
+    infile->AddListFile(myInputLists[i]);
+    se->registerInputManager(infile);
+  }
+
+  string outputDstFileName = "TrackCalo_" + to_string(segment) + "_dst.root";
+
+  string outputRecoDir = outDir + "/inReconstruction/" + to_string(runnumber) + "/";
+  string makeDirectory = "mkdir -p " + outputRecoDir;
+  system(makeDirectory.c_str());
+  string outputDstFile = outputRecoDir + outputDstFileName;
+
+  std::cout << "Output dst file: " << outputDstFile << std::endl;
+
+  Global_Reco();
+
+  std::string geofile = CDBInterface::instance()->getUrl("Tracking_Geometry");
+
+  Fun4AllRunNodeInputManager *ingeo = new Fun4AllRunNodeInputManager("GeoIn");
+  ingeo->AddFile(geofile);
+  se->registerInputManager(ingeo);
+
+  double dz_separation_0;
+  if (doDriftVelocityCorr)
+  {
+    dz_separation_0 = -0.13*2*105; // recommened drift velocity correction for Ar/CF4/N2 gas
+  }
+  else
+  {
+    dz_separation_0 = 0;
+  }
+  G4TPC::tpc_drift_velocity_reco = (8.0 / 1000) * 107.0 / 105.0 * (1+ dz_separation_0 / 2. / 105.); //cm/ns
+  G4TPC::ENABLE_MODULE_EDGE_CORRECTIONS = true;
+  //to turn on the default static corrections, enable the two lines below
+  G4TPC::ENABLE_STATIC_CORRECTIONS = true;
```

Comment:
Let's leave this to false right now. We don't understand the static corrections well enough at this point to have people running with them